### PR TITLE
Feature/avro-accumulo

### PR DESCRIPTION
**Pull request by zifeo**
Change serializer from Kryo to Avro

### CODE REVIEW

File: spark/src/main/scala/geotrellis/spark/io/index/hilbert/HilbertSpaceTimeKeyIndex.scala [hunk 1]
```diff
@@ -45,7 +41,7 @@ class HilbertSpaceTimeKeyIndex(
   val timeWidth = keyBounds.maxKey.temporalKey.time.getMillis - startMillis
   val temporalBinCount = math.pow(2, temporalResolution)
  
-  val chc = {
+  lazy val chc = {
```
Comment (echeipesh) [inline spark/src/main/scala/geotrellis/spark/io/index/hilbert/HilbertSpaceTimeKeyIndex.scala]: Good to know that java serialization does not chase after lazy vals.

Comment (zifeo) [inline spark/src/main/scala/geotrellis/spark/io/index/hilbert/HilbertSpaceTimeKeyIndex.scala]: I believe that `lazy` is exactly doing what you advise me, or at least it once did.

Comment (echeipesh) [inline spark/src/main/scala/geotrellis/spark/io/index/hilbert/HilbertSpaceTimeKeyIndex.scala]: This doesn't quite work. If the lazy val is accessed, I just put a `println(chc)` in the body, then java see will try to follow it (test will fail again with serialization errors). 

It appears that if you make it `@transiet lazy val` we get the behavior we're after.
Relevant SO post: http://stackoverflow.com/questions/27882307/how-does-serialization-of-lazy-fields-work

Comment (zifeo) [inline spark/src/main/scala/geotrellis/spark/io/index/hilbert/HilbertSpaceTimeKeyIndex.scala]: That make sense, I will push this change. Thanks!

**Comment by echeipesh** [comment]
+1

**Pull request merged**